nncf/torch/tensor_statistics/collectors.py

```diff
@@ -51,29 +61,49 @@ def reduce_max(x: NNCFTensor, axis: Union[int, tuple, list], keepdims: bool = Fa
     def abs(x: NNCFTensor) -> NNCFTensor:
         return PTNNCFTensor(torch.abs(x.tensor))
 
-    @staticmethod
-    def min(x1: NNCFTensor, x2: NNCFTensor) -> NNCFTensor:
-        return PTNNCFTensor(torch.min(x1.tensor, x2.tensor))
+    @classmethod
+    def min(cls, *args) -> NNCFTensor:
+        stacked = cls.stack(args)
+        return cls.reduce_min(stacked, axis=0, keepdims=False)
 
-    @staticmethod
-    def max(x1: NNCFTensor, x2: NNCFTensor) -> NNCFTensor:
-        return PTNNCFTensor(torch.max(x1.tensor, x2.tensor))
+    @classmethod
+    def max(cls, *args) -> NNCFTensor:
+        stacked = cls.stack(args)
+        return cls.reduce_max(stacked, axis=0, keepdims=False)
 
     @staticmethod
     def mean(x: NNCFTensor, axis: Union[int, tuple, list], keepdims=False) -> NNCFTensor:
         return PTNNCFTensor(x.tensor.mean(dim=axis, keepdim=keepdims))
 
     @staticmethod
     def median(x: NNCFTensor, axis: Union[int, tuple, list], keepdims=False) -> NNCFTensor:
-        return PTNNCFTensor(x.tensor.median(dim=axis, keepdim=keepdims))
+        # See https://github.com/pytorch/pytorch/issues/61582
+        if not isinstance(axis, int):
+            return PTNNCFTensor(torch.tensor(np.median(x.tensor.detach().cpu().numpy(), axis=axis, keepdims=keepdims)))
+        return PTNNCFTensor(torch.quantile(x.tensor, q=0.5, dim=axis, keepdim=keepdims).values)
 
-    @staticmethod
-    def masked_mean(x: NNCFTensor, axis: Union[int, tuple, list], mask: NNCFTensor, keepdims=False) -> NNCFTensor:
-        raise NotImplementedError()
+    @classmethod
+    def masked_mean(cls, x: NNCFTensor, axis: Union[int, tuple], mask: NNCFTensor, keepdims=False) -> NNCFTensor:
+        if mask is None:
+            return cls.mean(x, axis=axis, keepdims=keepdims)
+        masked_x = np.ma.array(x.tensor.detach().cpu().numpy(), mask=mask.tensor)
+        result = np.ma.mean(masked_x, axis=axis, keepdims=keepdims).astype(masked_x.dtype)
+        if isinstance(result, np.ma.MaskedArray):
+            return PTNNCFTensor(torch.tensor(result.data))
+        return PTNNCFTensor(torch.tensor(result))
```

Comment:
Why do we use `numpy` in the `torch` implementation?

Reply:
It can be left as-is for now, this will be replaced by direct tensor ops anyway which have a correct, torch-only [implementation](https://github.com/openvinotoolkit/nncf/pull/2120/files#diff-7746d0efc38f06575ddc5ce118d41c5e80850aa03a470ef2adb6954f87a57c15R310) for such ops

Reply:
The problem is that torch masked tensor [is in the prototype stage](https://pytorch.org/docs/stable/masked.html) and lacks some functions like median.